src/main/java/io/buildpacks/example/sample/GreetingController.java

```diff
@@ -7,8 +7,14 @@
 @RestController
 public class GreetingController {
 
-    @GetMapping("/greet")
-    public String greet(@RequestParam(value = "name", defaultValue = "World") String name) {
-        return String.format("Hello, %s!", name);
```

Comment:
6. Unit Testing: It would be beneficial to include unit tests for both the `greeting()` and `throwError()` methods to ensure they function as expected and handle edge cases properly.